### DISCUSSION TITLE
Compiling to bytecode, e.g. WebAssembly?

### DISCUSSION THREAD

**Discussion by pkese** [fable-compiler/Fable#3681] [General]
I'm just being curious...

Would it ever be possible for Fable to target WebAssembly, or is Fable intended just for spewing out source code rather than bytecode?

**Comment by pkese** [comment]
In recent versions WebAssembly apparently got support for:
- garbage collection
- tail calls
- exception handling

... so there's fewer and fewer reasons why it couldn't be a proper compiler target apart for the fact that WebAssembly is a binary thing (and probably needs linking, etc).

**Comment by ncave** [comment]
@pkese It's already possible to target WebAssembly via Fable -> Rust -> WebAsm. See [this](https://github.com/ncave/fable-raytracer) as an example.

The other option to target WebAsm is using the .NET `wasm-tools` [workload](https://learn.microsoft.com/en-us/aspnet/core/blazor/tooling?view=aspnetcore-8.0&pivots=windows#net-webassembly-build-tools) (using .NET AOT).
